deployment/cdk/opensearch-service-migration/lib/historical-capture-stack.ts

```diff
@@ -1,62 +1,97 @@
 
+        // ECS Task Definition
         const historicalCaptureFargateTask = new FargateTaskDefinition(this, "historicalCaptureFargateTask", {
             memoryLimitMiB: 2048,
             cpu: 512
         });
 
-        let logstashConfigData: string = readFileSync(props.logstashConfigFilePath, 'utf8');
-        if (props.sourceEndpoint) {
-            logstashConfigData = logstashConfigData.replace("<SOURCE_CLUSTER_HOST>", props.sourceEndpoint)
-        }
-        logstashConfigData = logstashConfigData.replace("<TARGET_CLUSTER_HOST>", props.targetEndpoint + ":80")
-        // Temporary measure to allow multi-line env variable
-        logstashConfigData = logstashConfigData.replace(/(\n)/g, "PUT_LINE")
         // Create Historical Capture Container
-        const historicalCaptureImage = new DockerImageAsset(this, "historicalCaptureImage", {
-            directory: join(__dirname, "../../..", "docker/logstash-setup")
-        });
-
         const historicalCaptureContainer = historicalCaptureFargateTask.addContainer("historicalCaptureContainer", {
-            image: ContainerImage.fromDockerImageAsset(historicalCaptureImage),
+            image: ContainerImage.fromAsset(join(__dirname, "../../../..", "FetchMigration")),
             // Add in region and stage
-            containerName: "logstash",
-            environment: {"LOGSTASH_CONFIG": '' + logstashConfigData},
-            logging: LogDrivers.awsLogs({ streamPrefix: 'logstash-lg', logRetention: 30 })
+            containerName: "fetch-migration",
+            logging: LogDrivers.awsLogs({ streamPrefix: 'fetch-migration-lg', logRetention: 30 })
+        });
+
+        // Create DP pipeline config from template file
+        let dpPipelineData: string = readFileSync(props.dpPipelineTemplatePath, 'utf8');
+        dpPipelineData = dpPipelineData.replace("<SOURCE_CLUSTER_HOST>", props.sourceEndpoint);
+        dpPipelineData = dpPipelineData.replace("<TARGET_CLUSTER_HOST>", props.targetEndpoint);
+        // Base64 encode
+        let encodedPipeline = Buffer.from(dpPipelineData).toString("base64");
+
+        // Create secret using Secrets Manager
+        const dpPipelineConfigSecret = new SMSecret(this, "dpPipelineConfigSecret", {
+            secretName: `${historicalCaptureFargateTask.family}-${historicalCaptureContainer.containerName}-pipelineConfig`,
+            secretStringValue: SecretValue.unsafePlainText(encodedPipeline)
         });
+        // Add secret to container
+        historicalCaptureContainer.addSecret("INLINE_PIPELINE", 
+            ECSSecret.fromSecretsManager(dpPipelineConfigSecret)
+        );
 
-        // Create Fargate Service
-        const historicalCaptureFargateService = new FargateService(this, "historicalCaptureFargateService", {
-            cluster: ecsCluster,
-            taskDefinition: historicalCaptureFargateTask,
-            desiredCount: 1
+        // The Migration Domain SG disallows all outgoing, but this prevents the ECS run task
```

Comment:
I wasn't aware of this as I haven't had issue with the other docker ECS services. Are you using a public ECR image versus a public docker image, which is the reason this is happening?

Reply:
The reason this is happening is because the `ecs run task` command is [executed via the Migration Console container](https://github.com/opensearch-project/opensearch-migrations/pull/346#discussion_r1355760947), which has this SG set on it. Kicking off an ECS task [requires outbound access to ECR over the public internet](https://repost.aws/knowledge-center/ecs-unable-to-pull-secrets).

The other ECS services don't run into this issue because their corresponding commands are executed on your developer host (which does not have such an outbound restriction set). Once the container is bootstrapped, this access is not needed.

Reply:
Good to know, thanks for the info 👍